### ISSUE TITLE
A query on the plotting?

### ISSUE THREAD

**Issue by rdstern** [IDEMSInternational/R-Instat#4397]
This is working well.

I have been doing PICSA-type graphs, with the year as the x-axis.  And various statistics plotted as the Y.  
1. For the trainer of trainers workshop I have 4 graphs on a page.  It will be single stations for the other workshops.
2. I use line plots, together with the points.  That's 2 layers.
3. Then I wanted to add the median for each station.  That's a straight line, but it is another layer, because it is different median for each station.
4. I did this by getting the median, and putting it back at the original level.  It works fine like that.
5. And doing it for multiple different statistics works great too.  I just change the Y variable and also edit the extra layer.  It works very quickly.  I like it.
6. I also found another feature I like.  I usually plot the last graph in the R viewer, because I can then copy to the clipboard and the resolution is better.  Then sometimes I would like to keep that graph.  So I simply rename the last_graph, using Prepare > R Objects > Rename.  Works fine.

Now my questions:
1. If I (instead) got the summary statistics at the Station level, i.e. not copied back to the original level, then could I use that instead?  I assume not, because the line plot needs the x to be the years and they would not be there?
2. What is the option to apply to all layers?  I haven't used it.  Is that relevant to the question above?

**Comment by dannyparsons** [comment]
> If I (instead) got the summary statistics at the Station level, i.e. not copied back to the original level, then could I use that instead?
Yes, but only using the General Graphics dialog, and this does relate to the apply on all layers checkbox.

 I used survey data.
1. added a layer of points: survey data; x = Village., y = Yield. **unchecked apply on all layers**.
2. added a layer of points survey_by_Village. data; x = Village., y = median_Yield (and made the points red so I could see the layers)

This then worked producing the code:
`last_graph <- ggplot2::ggplot() + ggplot2::geom_point(mapping=ggplot2::aes(x=Village., y=Yield), data=survey) + ggplot2::geom_point(mapping=ggplot2::aes(x=Village., y=median_Yield), data=survey_by_Village., colour="red")`

**Comment by dannyparsons** [comment]
Note: if you don't uncheck apply on all layers it doesn't work because it tries to use the summary data frame for both layers. I can see why it does that but not sure it's the most sensible behaviour. I think it should probably work without unchecking apply on all layers and when you add a second layer and change the data frame it should be able to work out which data frame is for which layer.
